### ISSUE TITLE
[Feature Request] Download button in long-press menu of library items

### ISSUE THREAD

**Issue by github-nicolas-stadler** [jmshrv/finamp#374]
Hey, for the sake of convenience, I would like to suggest some changes to the way the app manages downloaded tracks:
- Add a "Download" button to the long-press menu of library items (artists, albums, playlists, genres, tracks).
- Make said item "tagged" as downloaded instead of just being a one-time download. Items that are tagged as downloaded should managed themselves until I untag them. For example, if I add/remove/modify tracks in a downloaded playlist, the relevant files should be downloaded/removed/updated automatically so that I retain a locally accessible copy of the current state of that playlist (same logic with artists/albums and so on).

Thanks for the great app!

**Comment by Chaphasilor** [comment]
This depends on #213 so it might take a while. But it's a very nice idea!

**Comment by jmshrv** [comment]
Those are nice ideas, the first one would be very easy to implement and the second one has been a massive feature in the back of my head since I started working on this app 🙃

I really don't want to touch the download infrastructure until #213 is done, so yeah this will take a while. That bullet point is also a duplicate of #38, so I'll rename this issue to focus on the first one

**Comment by Chaphasilor** [comment]
This will be part of the redesign beta, courtesy of @Komodo5197 :D

**Comment by Chaphasilor** [comment]
@github-nicolas-stadler the download system has been improved as part of the redesign beta. You can check it out here: https://github.com/jmshrv/finamp/releases/tag/0.9.2-beta

Playlists will be refreshed with the current changes on app startup, and I'll work on automatically syncing changes when editing a playlist or creating a new one!

I'll close this issue for now since it should be resolved.
